stylelint-polaris/index.js

```diff
@@ -242,6 +282,16 @@ module.exports = {
       },
       depth: {
         'function-disallowed-list': [/([\w-]+\.)?shadow/],
+        'declaration-property-unit-disallowed-list': [
+          {
+            'box-shadow': ['px', 'rem', 'em'],
+          },
+          {severity: 'warning'},
+        ],
+        'property-disallowed-list': [
+          ['text-shadow', 'drop-shadow'],
+          {severity: 'warning'},
+        ],
```

Comment:
I've removed `drop-shadow` from the list since that's actually a filter function already rejected in the `function-disallowed-list` for depth. There's only [one instance](https://github.com/Shopify/web/blob/22175d78b38165d08e89e0aa941e5eb02ef16d6d/app/sections/Analytics/LiveView/styles/mixins.scss#L4) of `text-shadow` in `web` and the mixin is actually unused 🤔 I think we should warn against the use `text-shadow` because it's most often an escape hatch for mitigating inaccessible color contrast.